src/pilot/assign_role.py

```diff
@@ -850,6 +850,16 @@ def assign_role(ip_mac_service_tag, node_uuid, role_index, os_volume_size_gb,
               'path': '/properties/capabilities'}]
     ironic_client.node.update(node_uuid, patch)
 
+    # Detecting BOSS Card and find the boss device size
```

Comment:
We should only execute this code if the user has not specified os_volume_size on the command line.

Reply:
okay.